src/ui/public/index_patterns/_get_template_index_patterns.js

```diff
@@ -0,0 +1,21 @@
+import { map, flatten, pluck } from 'lodash';
+
+const getIndexPatternsFromResponse = json => {
+  return map(flatten(pluck(json, 'index_patterns')));
```

Comment:
I think it depends on whether or not there is some difference in selecting the `logstash-*` index pattern defined by template A or template B. If the end result is the same, then I think it makes sense to search the index pattern values (ignoring the template names), and to de-dupe identical index patterns in the UI.

But if the associated template of an index pattern does make a difference, then can we identify what that difference is? Once we understand how users will choose between template A and template, then we'll know which information to surface to the user to support that decision.